hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-globalpolicygenerator/src/main/java/org/apache/hadoop/yarn/server/globalpolicygenerator/GlobalPolicyGenerator.java

```diff
@@ -290,7 +291,9 @@ private String getHostName(Configuration config)
 
   public static void main(String[] argv) {
     try {
-      startGPG(argv, new YarnConfiguration());
+      YarnConfiguration conf = new YarnConfiguration();
+      new GenericOptionsParser(conf, argv);
```

Comment:
I'm not familiar with this but this use looks not complete.
Most other places use the getRemainingArgs().

Reply:
Thank you very much for helping review the code! When I first saw `YARN-11433`, my thoughts were aligned with yours, and I also had concerns about whether such an implementation was necessary.

I noticed `YARN-3255, where RM, NM, JobHistoryServer, and WebAppProxyServer's main() should support generic options.`  Other components in YARN, such as `RM`, `NM`, and `JobHistoryServer`, have implemented similar logic. I believe `Router` and `GPG` should be able to implement this logic as well.